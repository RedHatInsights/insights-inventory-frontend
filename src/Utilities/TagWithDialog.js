import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { TagCount } from '@redhat-cloud-services/frontend-components/TagCount';
import { loadTags, toggleTagModal } from '../store/actions';

const TagWithDialog = ({ count, systemId }) => {
  const dispatch = useDispatch();
  const triggerLoadTags = (systemId, count) => {
    if (systemId) {
      dispatch(toggleTagModal(true));
      dispatch(loadTags(systemId, undefined, undefined, count));
    }
  };

  return (
    <span
      onClick={(e) => e.stopPropagation()}
      className="ins-c-inventory__list-tags"
      data-ouia-component-id={`${systemId}-tag-button`}
    >
      <TagCount
        count={count}
        onTagClick={() => triggerLoadTags(systemId, count)}
      />
    </span>
  );
};

TagWithDialog.propTypes = {
  count: PropTypes.number,
  loadTags: PropTypes.func,
  systemId: PropTypes.string,
};

export default TagWithDialog;
