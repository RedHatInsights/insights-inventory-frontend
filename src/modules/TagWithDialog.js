import React from 'react';
import AsyncInventory from './AsyncInventory';
import TagWithDialogCmp from '../Utilities/TagWithDialog';

const BaseTagWithDialog = (props) => <AsyncInventory {...props} component={TagWithDialogCmp}  />;

const TagWithDialog = React.forwardRef((props, ref) => <BaseTagWithDialog {...props} innerRef={ref} />);

export default TagWithDialog;
