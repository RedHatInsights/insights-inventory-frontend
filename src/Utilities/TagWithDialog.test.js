/* eslint-disable react/display-name */
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import configureStore from 'redux-mock-store';
import { createPromise as promiseMiddleware } from 'redux-promise-middleware';
import TagWithDialog from './TagWithDialog';

jest.mock('../store/actions', () => {
  const actions = jest.requireActual('../store/actions');
  const ACTION_TYPES = jest.requireActual('../store/action-types');
  return {
    __esModule: true,
    ...actions,
    loadTags: () => ({
      type: ACTION_TYPES.LOAD_TAGS,
      payload: () => Promise.resolve([]),
    }),
  };
});

describe('EntityTable', () => {
  let mockStore;
  beforeEach(() => {
    mockStore = configureStore([promiseMiddleware()]);
  });

  describe('DOM', () => {
    it('should render with count', () => {
      const store = mockStore({});
      const view = render(<TagWithDialog store={store} count={10} />);

      expect(
        screen.getByRole('button', {
          name: '10',
        })
      ).toBeVisible();
      expect(view.asFragment()).toMatchSnapshot();
    });
  });

  describe('API', () => {
    it('should NOT call actions', async () => {
      const store = mockStore({});
      render(<TagWithDialog store={store} count={10} />);

      await userEvent.click(
        screen.getByRole('button', {
          name: '10',
        })
      );
      const actions = store.getActions();
      expect(actions.length).toBe(0);
    });

    it('should call actions', async () => {
      const store = mockStore({});
      render(<TagWithDialog store={store} count={10} systemId="something" />);

      await userEvent.click(
        screen.getByRole('button', {
          name: '10',
        })
      );
      const actions = store.getActions();
      expect(actions.length).toBe(3);
    });
  });
});
