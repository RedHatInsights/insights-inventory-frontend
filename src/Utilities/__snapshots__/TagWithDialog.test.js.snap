// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityTable DOM should render with count 1`] = `
<DocumentFragment>
  <span
    class="ins-c-inventory__list-tags"
    data-ouia-component-id="undefined-tag-button"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-plain ins-c-tag-count"
      data-ouia-component-id="OUIA-Generated-Button-plain-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1.5em"
        role="img"
        style="vertical-align: -0.1875em;"
        viewBox="0 0 512 512"
        width="1.5em"
      >
        <path
          d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
        />
      </svg>
      <span
        class="ins-c-tag__text"
      >
        10
      </span>
    </button>
  </span>
</DocumentFragment>
`;
